apps/s_client.c

```diff
@@ -2196,6 +2204,17 @@ int s_client_main(int argc, char **argv)
 #endif /* OPENSSL_NO_DTLS */
         sbio = BIO_new_socket(sock, BIO_NOCLOSE);
 
+    if (tfo) {
+        BIO_ADDRINFO *bai;
+
+        if (!BIO_lookup_ex(host, port, BIO_LOOKUP_CLIENT, socket_family,
+                           socket_type, protocol, &bai)) {
+            ERR_print_errors(bio_err);
+            goto shut;
+        }
+        (void)BIO_set_conn_address(sbio, BIO_ADDRINFO_address(bai));
+        BIO_ADDRINFO_free(bai);
+    }
```

Comment:
The duplicate BIO_lookup_ex call here is inefficient and also not fully correct, because there might be multiple addresses returned and you do not know here which one was actually used by the socket. Could you please refactor init_client to return the used address? Or handle it somehow else?